idaes/generic_models/properties/core/pure/tests/test_electrolyte.py

```diff
@@ -0,0 +1,84 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2020, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Tests for methods from electrolyte library
+
+Authors: Andrew Lee
+"""
+
+import pytest
+import types
+
+from pyomo.environ import \
+    ConcreteModel, Block, Expression, value, Var, units as pyunits
+from pyomo.common.config import ConfigBlock
+from pyomo.util.check_units import assert_units_equivalent
+
+from idaes.generic_models.properties.core.pure.electrolyte import \
+    relative_permittivity_constant
+from idaes.core.util.misc import add_object_reference
+from idaes.core.property_meta import PropertyClassMetadata
+
+
+@pytest.fixture()
+def frame():
+    m = ConcreteModel()
+
+    # Create a dummy parameter block
+    m.params = Block()
+
+    m.params.config = ConfigBlock(implicit=True)
+    m.params.config.parameter_data = {
+        "relative_permittivity_liq_comp": 101}
+
+    m.meta_object = PropertyClassMetadata()
+    m.meta_object.default_units["temperature"] = pyunits.K
+    m.meta_object.default_units["mass"] = pyunits.kg
+    m.meta_object.default_units["length"] = pyunits.m
+    m.meta_object.default_units["time"] = pyunits.s
+    m.meta_object.default_units["amount"] = pyunits.mol
+
+    def get_metadata(self):
+        return m.meta_object
+    m.get_metadata = types.MethodType(get_metadata, m)
+    m.params.get_metadata = types.MethodType(get_metadata, m.params)
+
+    # Add necessary parameters to parameter block
+    m.params.temperature_ref = Var(initialize=298.15, units=pyunits.K)
+    m.params.pressure_ref = Var(initialize=1e5, units=pyunits.Pa)
+
+    # Create a dummy state block
+    m.props = Block([1])
```

Comment:
Just curious- what is the purpose of providing this index of 1 here? I am also trying to understand when/why it would be useful to implement indexed blocks.

Reply:
Within IDAES, `StateBlocks` are always indexed by time (and sometimes by space), thus we generally index the `StateBlock` during testing to make sure we replicate the actual conditions it will be used under. The reason for indexing is to keep each point in space and time separate, and to avoid having to index all the properties individually.